### ISSUE TITLE
Deep copy in Version::marshalData() corrupts in-memory Document object

### ISSUE THREAD

**Issue by pmuskovac** [pimcore/pimcore#7628]
# Bug Report
This appeared when upgrading to 6.8 (v6.8.5) using both PHP 7.3 and 7.4.
We have some custom EventHandlers that do validation and data postprocessing on Relations in Pages.
In `pimcore.document.preUpdate` the Page object passed to the event looks fine.
In `pimcore.document.postUpdate` we run into a strange crash:
`[Error] Call to undefined method Pimcore\Model\Element\ElementDescriptor::getProperty()`

Condensing the code we run, it's
`DocumentEvent::getDocument()->getEditables()['mainArea:1.relatedObject']->element->getProperty('x')`

Using xdebug to look at the data when the Page is saved:
'mainArea:1.relatedObject' is an instance of `Pimcore\Model\Document\Editable\Relation`
- before Version::save():
 Relation::$element is an instance of `Pimcore\Model\DataObject\Article`
- after Version::save():
 Relation::$element is an instance of `Pimcore\Model\Element\ElementDescriptor`

so it seems like DeepCopy->copy() is also modifying the source object, and not just the clone, resulting in a corrupted Page object being passed to `pimcore.document.postUpdate`.

### Expected behavior
identical objects are passed to `pimcore.document.preUpdate` and `pimcore.document.postUpdate`

### Actual behavior
the Document object passed to `pimcore.document.postUpdate` contains marshalled relations.

### Steps to reproduce
- create a Page with Relation Editables on the page.
- create an EventHandler for `pimcore.document.postUpdate` that includes the code example below

### Code example
```php
        foreach ($event->getDocument()->getEditables() as $editable) {
            if ($editable instanceof Relation) {
                $editable->getElement()->getProperties();
            }
        }
```

**Comment by dvesh3** [comment]
Fixed by #7706

**Comment by jdreesen** [comment]
Maybe the milestone of this issue should be adjusted, as #7706 was merged into the `6.9` branch.

**Comment by dvesh3** [comment]
@jdreesen good catch! done 😉 thanks a lot!
